src/vsr/replica.zig

```diff
@@ -766,6 +766,12 @@ pub fn ReplicaType(
             tracer.flush();
         }
 
+        // Pings are used:
+        //  - By clients, to learn about the current view.
+        //  - By replicas, to synchronise cluster time and to probe for network connectivity.
+        //
+        // In the second case we avoid setting the view to make sure pings can still be sent
+        // during view changes.
```

Comment:
We could additionally enforce this invariant in `vsr.zig` (`invalid_ping()`/`invalid_pong()`). Those functions are used the reject messages before they reach `on_ping()`/`on_pong()`.